### ISSUE TITLE
x:Bind not Binding

### ISSUE THREAD

**Issue by miniksa** [microsoft/terminal#11767]
Per @Austin-Lamb ... We should avoid Binding everywhere we can and use x:Bind instead.

This represents not forgetting about this by tomorrow morning as it popped in my head and reminded me as I tried to sleep.

**Comment by zadjii-msft** [comment]
protip: find in all files for `{Binding `. That'll get you:
![image](https://user-images.githubusercontent.com/18356694/141977555-c4f21991-4ece-47f9-84a4-e64cbce2fcc6.png)


`7 matches across 5 files`

**Comment by miniksa** [comment]
I was told also that x:Bind is not 100% feature compatible with Binding... but that x:Bind should be strongly preferred wherever it can be used.

**Comment by Austin-Lamb** [comment]
Thanks for filing this, I was meaning to but have been busy.  Yeah, {x:Bind} is preferred for performance over {Binding}.  x:Bind lives in hotter code in XAML's binary so it will be loaded off disk more quickly, and it is optimized for better CPU performance by not doing as much "Reflection" at runtime (in C++ it's not really reflection, it's IXamlMetadataProvider).  Note that x:Bind generates code into your app so it will cause your binaries to grow, but the amount should be marginal for how few bindings Terminal has.

One 'gotcha' with x:Bind (as noted on [its docs page here](https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/x-bind-markup-extension)) is that x:Bind defaults to OneTime mode.  This can be good for perf (which is why it's the default) but it can be easy to overlook when converting a Binding.  You can either manually set the Mode in each x:Bind, or if you know you want to do that for every binding in a tree/file you can use x:DefaultBindMode).

It's been a while since I looked at this, but it would seem that x:Bind may not support ElementName bindings.  You may need to check with the owners of bindings in XAML to see if there's a way to do what you're doing there but using x:Bind (perhaps via function bindings, I'm just not sure how you'd do change notifications correctly in a case like biding to the Slider's Value).

**Comment by smeet07** [comment]
Does this still need to be worked on? I

**Comment by Bhavika42** [comment]
Hey! I'd love to work on this issue as my contribution.  
Could you please confirm if it's still relevant and assign it to me? 😊

**Comment by zadjii-msft** [comment]
I'd assume that it does need to be worked on ; feel free to get started on it if you'd like. We typically don't actually assign issues, since the Assigned-to field across the repo is usually pretty stale 😕

**Comment by jamespack** [comment]
> I was told also that x:Bind is not 100% feature compatible with Binding... but that x:Bind should be strongly preferred wherever it can be used.

I dont think it can supports relative binding and binding to framework elements.

**Comment by shivam-dhir** [comment]
Hi, I would love to work on this issue

**Comment by lekithkishore** [comment]
It would be my pleasure to solve this . Could i be assigned this or pherhaps given some pointers

**Comment by zadjii-msft** [comment]
I mean, if you read the thread, I think we've pretty much discussed everything there is to know here. Including the fact that 

> We typically don't actually assign issues

**Comment by lekithkishore** [comment]
> > We typically don't actually assign issues

Well sorry to ask this but would not being assigned make our contributions invisible?

**Comment by zadjii-msft** [comment]
Uh, no? Contribution is counted when PRs are merged. No one cares about the assigned-to field. Code wins.

**Comment by lekithkishore** [comment]
Well looks like i have learned something new . Thanks for that
